apiServices/userServices.ts

```diff
@@ -52,3 +52,13 @@ function createUploadData(uri: string, body?: { [key: string]: string }) {
 
   return formData
 }
+
+export async function addLover(userId: string): Promise<IUser> {
+  const res = await authFetch<IUser>('POST', 'api/users/relationship', {
+    user_id: userId
+  })
+  if (res.status !== 200) {
```

Comment:
Because of this if-statement, this function describes two different workflows.

I anticipated that because **some business reason**, it is more likely for us only to read a single workflow at a time. Also, because **some business reason** I can imagine that the expected change will be to introduce a new workflow for a new specific case. 

If we change this if to factory and polymorph, we will have two workflows written in a different file in a different class. There is a higher chance that this will suit our future need.